evm/src/cpu/kernel/interpreter.rs

```diff
@@ -181,6 +196,137 @@ impl<'a> Interpreter<'a> {
         result
     }
 
+    /// Initialize interpreter state given `GenerationInputs`. This assumes that the
+    /// loaded code is the KERNEL code.
```

Comment:
Not sure to understand the second sentence. It's not really assuming per se as it's explicitely hardcoding the use of KERNEL code.